src/client/wire_protocol/operations.rs

```diff
@@ -0,0 +1,309 @@
+use bson::Document as BSON;
+use bson;
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use client::wire_protocol::header::{Header, OpCode};
+use std::io::{Read, Write};
+use std::mem;
+use std::result::Result::{Ok, Err};
+
+/// Gets the length of a BSON document.
+//
+/// # Arguments
+///
+/// `bson` - The BSON document to find the length of.
+///
+/// # Return value
+///
+/// Returns the number of bytes in the serialized BSON document.
+fn length_of_bson(bson: &BSON) -> Result<i32, String> {
```

Comment:
Yeah, I thought about that, and I probably should have done it. I was in a rush when I added this since it was a fix to the bug where the server was expecting a message that was too long.